rules/ast/functions.go

```diff
@@ -498,6 +499,43 @@ func derivImpl(timestamp clientmodel.Timestamp, args []Node) interface{} {
 	return resultVector
 }
 
+// === histogram_quantile(k ScalarNode, vector VectorNode) Vector ===
```

Comment:
"You're doing a implicit sum there, which I think would cause confusion. No other function does that." 

That was my original point. If `histogram_quantile` supports a `by` clause at all, it will implicitly sum up for the implied aggregation. The original proposal is always explicit about the aggregation.

Given the confusion already caused, I would suggest to go with keeping `histogram_quantile` a function for now, with the same fundamental semantics as `topk`, and then change things consistently across the language when we introducy `by` clauses for the existing functions and binary operators. Otherwise, `histogram_quantile` would be the only function that has that "new style" semantics already.